mono/mini/debugger-agent.c

```diff
@@ -6264,8 +6788,220 @@ buffer_add_value_full (Buffer *buf, MonoType *t, void *addr, MonoDomain *domain,
 		NOT_IMPLEMENTED;
 	}
 }
-
-static void
+#else
+static void buffer_add_value_full(Buffer *buf, MonoType *t, void *addr, MonoDomain *domain,
+	gboolean as_vtype, GHashTable *parent_vtypes)
+{
+	MonoObject *obj;
+	gboolean boxed_vtype = FALSE;
+
+	if (il2cpp_type_is_byref(t))
+	{
+		if (!(*(void**)addr))
+		{
+			/* This can happen with compiler generated locals */
+			//printf ("%s\n", mono_type_full_name (t));
+			buffer_add_byte(buf, VALUE_TYPE_ID_NULL);
+			return;
+		}
+		g_assert(*(void**)addr);
+		addr = *(void**)addr;
+	}
+
+	if (as_vtype)
+	{
+		switch (il2cpp_type_get_type(t))
+		{
+			case MONO_TYPE_BOOLEAN:
+			case MONO_TYPE_I1:
+			case MONO_TYPE_U1:
+			case MONO_TYPE_CHAR:
+			case MONO_TYPE_I2:
+			case MONO_TYPE_U2:
+			case MONO_TYPE_I4:
+			case MONO_TYPE_U4:
+			case MONO_TYPE_R4:
+			case MONO_TYPE_I8:
+			case MONO_TYPE_U8:
+			case MONO_TYPE_R8:
+			case MONO_TYPE_I:
+			case MONO_TYPE_U:
+			case MONO_TYPE_PTR:
+				goto handle_vtype;
+				break;
+			default:
+				break;
+		}
+	}
+
+	switch (il2cpp_type_get_type(t))
+	{
+		case MONO_TYPE_VOID:
+			buffer_add_byte(buf, il2cpp_type_get_type(t));
+			break;
+		case MONO_TYPE_BOOLEAN:
+		case MONO_TYPE_I1:
+		case MONO_TYPE_U1:
+			buffer_add_byte(buf, il2cpp_type_get_type(t));
+			buffer_add_int(buf, *(gint8*)addr);
+			break;
+		case MONO_TYPE_CHAR:
+		case MONO_TYPE_I2:
+		case MONO_TYPE_U2:
+			buffer_add_byte(buf, il2cpp_type_get_type(t));
+			buffer_add_int(buf, *(gint16*)addr);
+			break;
+		case MONO_TYPE_I4:
+		case MONO_TYPE_U4:
+		case MONO_TYPE_R4:
+			buffer_add_byte(buf, il2cpp_type_get_type(t));
+			buffer_add_int(buf, *(gint32*)addr);
+			break;
+		case MONO_TYPE_I8:
+		case MONO_TYPE_U8:
+		case MONO_TYPE_R8:
+			buffer_add_byte(buf, il2cpp_type_get_type(t));
+			buffer_add_long(buf, *(gint64*)addr);
+			break;
+		case MONO_TYPE_I:
+		case MONO_TYPE_U:
+			/* Treat it as a vtype */
+			goto handle_vtype;
+		case MONO_TYPE_PTR: {
+			gssize val = *(gssize*)addr;
+
+			buffer_add_byte(buf, il2cpp_type_get_type(t));
+			buffer_add_long(buf, val);
+			break;
+		}
+						handle_ref:
```

Comment:
What's this doing here?

Reply:
The goto label inside the switch? That was there previously, the white space changed I guess.  In addition to making us all cringe, I'm not sure this will work with our code.  We're letting tests drive development, so we haven't hit this yet.

Reply:
I don't know about the whitespace issue. In my editor I only see one tab on line 6874.